packages/venia-ui/lib/components/Dialog/dialog.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { bool, func, shape, string, object } from 'prop-types';
+import { Form } from 'informed';
+import { X as CloseIcon } from 'react-feather';
+
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+
+import Icon from '../Icon';
+import { Modal } from '../Modal';
+import defaultClasses from './dialog.css';
+
+/**
+ * The Dialog component shows its children content in a dialog,
+ * encoding the look-and-feel and behavior in one place for consistency across the app.
+ *
+ * @typedef Dialog
+ * @kind functional component
+ *
+ * @param {Object}  props
+ * @param {BOolean} props.areButtonsDisabled - A toggle for whether the buttons are enabled.
+ * @param {Object}  props.classes - A set of class overrides to apply to elements.
+ * @param {String}  props.cancelText - The text to display on the Dialog cancel button.
+ * @param {String}  props.confirmText - The text to display on the Dialog confirm button.
+ * @param {Object}  props.formProps - Props to apply to the internal form. @see https://joepuzzo.github.io/informed/?path=/story/form--props.
+ * @param {Boolean} props.isModal - Determines behavior of clicking on the mask. False cancels Dialog.
+ * @param {Boolean} props.isOpen - Whether the Dialog is currently showing.
+ * @param {Func}    props.onCancel - A function to call when the user cancels the Dialog.
+ * @param {Func}    props.onConfirm - A function to call when the user confirms the Dialog.
+ * @param {String}  props.title - The title of the Dialog.
+ */
+const Dialog = props => {
+    const {
+        areButtonsDisabled,
```

Comment:
I see what your intent is here, but this doesn't feel right. If i were writing this component, I might have called this one `shouldDisableButtons`, since props are essentially recommendations that the component uses to determine final state.

```es6
// disable the mask if a parent says we should
const maskIsDisabled = isModal || shouldDisableButtons
```

Reply:
Fixed in https://github.com/magento/pwa-studio/pull/2365/commits/d5841abb26d7a014b3bf9d50bdd5fe38c4dc0b77.

Reply:
Fun side note: I changed this and a snapshot that I wasn't expecting to change changed and I was able to fix the bug before anyone else ever saw it - yay snapshots!